apps/services/user-notification/src/app/modules/notifications/notifications.service.ts

```diff
@@ -341,4 +341,48 @@ export class NotificationsService {
       )
     }
   }
+
+  async getUnreadNotificationsCount(user: User): Promise<{unreadCount:number}> {
+    try {
+      const unreadCount = await this.notificationModel.count({
+        where: {
+          recipient: user.nationalId,
+          read: false,
+        },
+      })
+      return { unreadCount }
+    } catch (error) {
+      this.logger.error('Error getting unread notifications count:', error)
+      throw new InternalServerErrorException('Error getting unread notifications count')
+    }
+  }
+
+  async getUnseenNotificationsCount(user: User): Promise<{unseenCount:number}> {
+    try {
+      const unseenCount = await this.notificationModel.count({
+        where: {
+          recipient: user.nationalId,
+          seen: false,
+        },
+      })
+      return { unseenCount }
+    } catch (error) {
+      this.logger.error('Error getting unseen notifications count:', error)
+      throw new InternalServerErrorException('Error getting unseen notifications count')
+    }
+  }
+
+  async markAllAsSeen(user: User): Promise<void> {
+    try {
+      await this.notificationModel.update(
+        { seen: true },
+        { where: { recipient: user.nationalId, seen: false } }
```

Comment:
I'm not 100% certain how `CreateOptional` works but as `seen` is declared that way, is it then possible that `null` entries for `seen` exist?

Should this be something like `{ where: { recipient: user.nationalId, seen: { $not: true } }`? (I don't remember the ORM too well)